### ISSUE TITLE
Server Builds are inconsistent (URGENT help needed)

### ISSUE THREAD

**Issue by fxmb** [medusajs/medusa#5618]
Since last week my server has some weird inconsistencies: It sometimes uses old file versions that were overwritten, even after a fresh build (yes I remove /dist and even /build) before rebuilding.

```
✔ Server is ready on port: 9000 – 8ms
::1 - - [06/Nov/2023:11:00:52 +0000] "OPTIONS /store/carts/cart_01HEJ4JKW7FXW5V7GQ8R66KECE/line-items HTTP/1.1" 204 0 "http://localhost:8000/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36"
::1 - - [06/Nov/2023:11:00:52 +0000] "POST /store/carts/cart_01HEJ4JKW7FXW5V7GQ8R66KECE/line-items HTTP/1.1" 200 7358 "http://localhost:8000/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36"
🚩 **info:    Processing cart.updated which has 0 subscribers**
::1 - - [06/Nov/2023:11:00:55 +0000] "POST /store/carts/cart_01HEJ4JKW7FXW5V7GQ8R66KECE/line-items HTTP/1.1" 200 7358 "http://localhost:8000/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36"
🚩 **info:    Processing cart.updated which has 1 subscribers**
```

You can see it here: First the cart has 0 subscribers, then all of a sudden 1 just once call later. Trigger the same event twice (e.g. Cart Updated) and the subscriber runs different versions of the subscriber (e.g. an older one) on each function invocaition...

Also if I e.g. add a logger.info("TEST") to that subscriber and rebuild, sometimes it will include the Log, sometimes it just builds (and logs) nothing on a consecutive build.

This is my start command:

`yarn build && yarn start:prod`

and 

```
"build": "yarn run clean && tsc -p tsconfig.json",
"start:prod": "medusa migrations run && medusa start",
```

it doesnt make any sense to me...

Is this a known issue? Maybe also not Medusa specific but an Express issue 

 Basically:
```
info:    Processing cart.updated which has 0 subscribers
....
info:    Processing cart.updated which has 1 subscribers
....
info:    Processing cart.updated which has 0 subscribers 
```

I feel like this might have to do with Redis though... 

```
    "@medusajs/cache-redis": "1.9.0",
    "@medusajs/event-bus-local": "^1.9.0",
    "@medusajs/event-bus-redis": "1.8.10",
    "@medusajs/medusa": "1.17.2",
```

**Comment by fxmb** [comment]
Sometimes when I start my server it uses code that I wrote weeks ago but it is long gone and legacy... It doesnt even exist anymore in the codebase. This seems to only affect subscribers though.

The problem seems to only exist locally, not in production

All of these updates are one consecutive call of the same build.. It simply does not make sense:

<img width="553" alt="image" src="https://github.com/medusajs/medusa/assets/39501592/626579da-992f-4e2c-bada-b98f8e947889">

**Comment by fxmb** [comment]
Could this have anything to do with me using multiple local eventbuses? E.g. Redis and local? Could they be overwriting each other?

```
    "@medusajs/cache-inmemory": "^1.8.0",
    "@medusajs/cache-redis": "^1.9.0",
    "@medusajs/event-bus-local": "^1.9.0",
    "@medusajs/event-bus-redis": "1.8.3",
    "@medusajs/medusa": "1.17.4",
```
